src/sentry/snuba/discover.py

```diff
@@ -0,0 +1,230 @@
+from __future__ import absolute_import
+
+import six
+
+from copy import deepcopy
+from sentry.api.event_search import get_filter, resolve_field_list
+from sentry.utils.snuba import (
+    Dataset,
+    DISCOVER_COLUMN_MAP,
+    QUOTED_LITERAL_RE,
+    get_function_index,
+    raw_query,
+)
+
+ALLOWED_INTERNALS = set(["project_id", "event_id"])
+
+
+def resolve_column(col):
+    """
+    Resolve a public schema name to the discover dataset.
+    unknown columns are converted into tags expressions.
+    """
+    if not col:
+        return ""
+    if col.startswith("tags[") or col in ALLOWED_INTERNALS or QUOTED_LITERAL_RE.match(col):
+        return col
+    return DISCOVER_COLUMN_MAP.get(col, u"tags[{}]".format(col))
+
+
+def resolve_condition(cond):
```

Comment:
Is this the same code as `constrain_condition_to_dataset` copied over? It looks roughly the same to me at a glance. If not, what's changed and can we reuse some of the implementation rather than have 2 versions?

Reply:
It is mostly the same code. The differences are the lack of a `dataset` parameter, and that `type:transaction` conditions are no longer dropped.

By duplicating the code I was biasing to making the current code easy to delete when this implementation takes over. Making a slightly different implementation also reduces risk while this approach is put into use by endpoints, as I won't accidentally break the existing code paths.

Reply:
Ok. Can you mark the parts that are intended to be deprecated as such? It's easier for people reading code / doing review to see the direction it's going in.